query/src/main/scala/filodb/query/exec/aggregator/RowAggregator.scala

```diff
@@ -120,33 +120,30 @@ object RowAggregator {
     */
   def apply(aggrOp: AggregationOperator, params: Seq[Any], schema: ResultSchema): RowAggregator = {
     val valColType = ResultSchema.valueColumnType(schema)
-    if (valColType == ColumnType.HistogramColumn) {
-      aggrOp match {
-        case Sum if isHistMax(valColType, schema) => HistMaxSumAggregator
-        case Sum   => HistSumRowAggregator
-        case Count => CountRowAggregator.hist
-        case _ => throw new IllegalArgumentException(s"The ${aggrOp} operation is not supported directly for histogram"
-          + s" types. Check if you need to resolve to the individual histogram buckets,"
-          + s"or calculate the rate and histogram_quantile before applying the aggregation. "
-          + s"If you have a genuine use case for this query, please get in touch.")
-      }
-    } else if (valColType == ColumnType.DoubleColumn) {
-      aggrOp match {
-        case Min => MinRowAggregator
-        case Max => MaxRowAggregator
-        case Sum => SumRowAggregator
-        case Count => CountRowAggregator.double
-        case Group => GroupRowAggregator
-        case Avg => AvgRowAggregator
-        case TopK => new TopBottomKRowAggregator(params(0).asInstanceOf[Double].toInt, false)
-        case BottomK => new TopBottomKRowAggregator(params(0).asInstanceOf[Double].toInt, true)
-        case Quantile => new QuantileRowAggregator(params(0).asInstanceOf[Double])
-        case Stdvar => StdvarRowAggregator
-        case Stddev => StddevRowAggregator
-        case CountValues => new CountValuesRowAggregator(params(0).asInstanceOf[String])
-      }
-    } else {
-      throw new UnsupportedOperationException(s"The ${aggrOp} operation is not supported for ${valColType}");
+    aggrOp match {
+      case Min if valColType == ColumnType.DoubleColumn => MinRowAggregator
+      case Max if valColType == ColumnType.DoubleColumn => MaxRowAggregator
+      case Sum if valColType == ColumnType.DoubleColumn => SumRowAggregator
+      case Sum if isHistMax(valColType, schema) => HistMaxSumAggregator
+      case Sum if valColType == ColumnType.HistogramColumn => HistSumRowAggregator
+      case Count if valColType == ColumnType.DoubleColumn => CountRowAggregator.double
+      case Count if valColType == ColumnType.HistogramColumn => CountRowAggregator.hist
+      case Group if valColType == ColumnType.DoubleColumn => GroupRowAggregator
+      case Avg if valColType == ColumnType.DoubleColumn => AvgRowAggregator
+      case TopK => new TopBottomKRowAggregator(params(0).asInstanceOf[Double].toInt, false)
+      case BottomK => new TopBottomKRowAggregator(params(0).asInstanceOf[Double].toInt, true)
+      case Quantile => new QuantileRowAggregator(params(0).asInstanceOf[Double])
+      case Stdvar if valColType == ColumnType.DoubleColumn => StdvarRowAggregator
+      case Stddev if valColType == ColumnType.DoubleColumn => StddevRowAggregator
```

Comment:
@vishramachandran I see `Group`, `Avg`, `StdVar` and `StdDev` are now restricted to column type double only. Would that be ok? I mentioned a couple of times [here](https://github.com/filodb/FiloDB/pull/1545#discussion_r1153543524) that we should try and minimize the blast radius by just fixing the count and sum for histograms only and let everything else continue to work the way it is. We already found a regression in count and luckily I found regression in topk/bottomk during my local testing. I would like to get your views. I'll be holding off my approval for the PR till then.Thanks

cc @kvpetrov @sandeep6189

Reply:
I agree on containing the blast radius and minimizing regressions. At the same time we need to send out a clearer error message that helps the user esp if the query is not meaningful. I don't see us supporting some of the aggregations (like min, max, stddev, stdvar etc) on histograms. Do you see that possibility ?

I think we are seeing this regression because of absence of unit tests esp for this function. Improving unit tests by running this function through various types could make it more solid, but it comes at the cost of increasing this PR scope.